internal/k8s/controller.go

```diff
@@ -3754,38 +3769,31 @@ func (lbc *LoadBalancerController) getEndpointsForPortFromEndpointSlices(endpoin
 		return nil, fmt.Errorf("no port %v in service %s", backendPort, svc.Name)
 	}
 
-	endpointSet := make(map[podEndpoint]struct{})
-	for _, endpointSlice := range endpointSlices {
-		for _, endpointSlicePort := range endpointSlice.Ports {
-			if *endpointSlicePort.Port == targetPort {
-				for _, endpoint := range endpointSlice.Endpoints {
-					if !*endpoint.Conditions.Ready {
-						continue
-					}
-					for _, endpointAddress := range endpoint.Addresses {
-						address := ipv6SafeAddrPort(endpointAddress, *endpointSlicePort.Port)
-						podEndpoint := podEndpoint{
-							Address: address,
-						}
-						if endpoint.TargetRef != nil {
-							parentType, parentName := lbc.getPodOwnerTypeAndNameFromAddress(endpoint.TargetRef.Namespace, endpoint.TargetRef.Name)
-							podEndpoint.OwnerType = parentType
-							podEndpoint.OwnerName = parentName
-							podEndpoint.PodName = endpoint.TargetRef.Name
-						}
-						endpointSet[podEndpoint] = struct{}{}
-					}
+	makePodEndpoints := func(port int32, epx []discovery_v1.Endpoint) []podEndpoint {
+		endpointSet := make(map[podEndpoint]struct{})
+
+		for _, ep := range epx {
+			for _, addr := range ep.Addresses {
+				address := ipv6SafeAddrPort(addr, port)
+				podEndpoint := podEndpoint{
+					Address: address,
 				}
+				if ep.TargetRef != nil {
```

Comment:
do we need unit tests to cover this if block?

Reply:
hi, line 3782, method `lbc.getPodOwnerTypeAndNameFromAddress(ep.TargetRef.Namespace, ep.TargetRef.Name)` is already covered.